File: product-service/src/main/java/com/intexsoft/webshop/productservice/repository/SubcategoryRepository.java
package com.intexsoft.webshop.productservice.repository;

import com.intexsoft.webshop.productservice.model.Subcategory;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface SubcategoryRepository extends JpaRepository<Subcategory, Long> {
    Optional<Subcategory> findByNameIgnoreCase(String name);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/repository/SubcategoryRepository.kt
package com.intexsoft.webshop.productservicekt.repository

import com.intexsoft.webshop.productservicekt.model.Subcategory
import org.springframework.data.jpa.repository.JpaRepository

interface SubcategoryRepository : JpaRepository<Subcategory, Long> {
    fun findByNameIgnoreCase(name: String): Subcategory?
}
